mbox/src/main/java/org/apache/james/mime4j/mboxiterator/MboxIterator.java

```diff
@@ -76,8 +78,12 @@ private MboxIterator(final File mbox,
         this.DECODER = charset.newDecoder();
         this.mboxCharBuffer = CharBuffer.allocate(MAX_MESSAGE_SIZE);
         this.mbox = mbox;
-        this.theFile = new FileInputStream(mbox);
-        this.byteBuffer = theFile.getChannel().map(FileChannel.MapMode.READ_ONLY, 0, theFile.getChannel().size());
+
+        try (FileChannel channel = FileChannel.open(mbox, StandardOpenOption.READ)) {
+            // Safe to keep buffer around because it does not depend on the channel.
+            this.byteBuffer = channel.map(FileChannel.MapMode.READ_ONLY, 0, channel.size());
```

Comment:
I never used that API but when will the resource be disposed then?

Reply:
The `ByteBuffer` is disposed of when garbage collected. There is no explicit way of releasing it:

From https://docs.oracle.com/en/java/javase/14/docs/api/java.base/java/nio/channels/FileChannel.html#map(java.nio.channels.FileChannel.MapMode,long,long) :
> The buffer and the mapping that it represents will remain valid until the buffer itself is garbage-collected.

Reply:
Thank you for the link and the explanation